### PR TITLE
Added parallel threads for 4 sensors

### PR DESCRIPTION
**Pull request by tomin11**
- added 4 boost threads

### CODE REVIEW

File: src/vl53l0x_driver.cpp [hunk 1]
```diff
@@ -103,18 +106,72 @@ int check_device_connection(int _file_descriptor) {
...
-                                          &SensorsRangingMeasurementData[i]);
-  sensor_msg_array[i].proximity =
-      float(SensorsRangingMeasurementData[i].RangeMilliMeter) / 1000.0;
-  sensor_msg_array[i].header.stamp = ros::Time::now();
-  std::string frame = "sensor";
-  sensor_msg_array[i].header.frame_id = frame + std::to_string(i + 1);
-  sensor_msg_array[i].field_of_view = 0.436332;
-  sensor_msg_array[i].min_range = 0.03;
-  sensor_msg_array[i].max_range = 2.0;
+void thread1() {
```
Comment (bhuvanchandradv) [inline src/vl53l0x_driver.cpp]: can you use a common method for the threads

File: src/vl53l0x_driver.cpp [hunk 2]
```diff
@@ -103,18 +106,72 @@ int check_device_connection(int _file_descriptor) {
...
+  while (ros::ok() and check_device_connection(pSensors[0]->fd)) {
+    VL53L0X_PerformSingleRangingMeasurement(pSensors[0],
+                                            &SensorsRangingMeasurementData[0]);
+    sensor_msg_array[0].proximity =
+        float(SensorsRangingMeasurementData[0].RangeMilliMeter) / 1000.0;
+    sensor_msg_array[0].header.stamp = ros::Time::now();
+    std::string frame = "sensor";
+    sensor_msg_array[0].header.frame_id = frame + std::to_string(1);
+    sensor_msg_array[0].field_of_view = 0.436332;
```
Comment (bhuvanchandradv) [inline src/vl53l0x_driver.cpp]: make the constants as #define

File: src/vl53l0x_driver.cpp [hunk 3]
```diff
@@ -136,13 +191,14 @@ int main(int argc, char **argv) {
...
         Sensor_Calibration(pSensors[i]);
       }
-      while (ros::ok() and check_device_connection(pSensors[0]->fd)) {
-        for (int i = 0; i < NUM_SENSORS; i++) {
-          Start_Ranging(i);
-          sensor_pub_array[i].publish(sensor_msg_array[i]);
-          ros::spinOnce();
-        }
-      }
+      ros::AsyncSpinner spinner(4); // Use 4 threads
```
Comment (bhuvanchandradv) [inline src/vl53l0x_driver.cpp]: we should be able to pass the number of sensors

Comment (tomin11) [reply src/vl53l0x_driver.cpp]: ok

File: src/vl53l0x_driver.cpp (hunk 1 above)
Comment (tomin11) [reply src/vl53l0x_driver.cpp]: ok

File: src/vl53l0x_driver.cpp (hunk 2 above)
Comment (tomin11) [reply src/vl53l0x_driver.cpp]: ok

File: src/vl53l0x_driver.cpp [hunk 4]
```diff
@@ -103,18 +109,21 @@ int check_device_connection(int _file_descriptor) {
...
-                                          &SensorsRangingMeasurementData[i]);
-  sensor_msg_array[i].proximity =
-      float(SensorsRangingMeasurementData[i].RangeMilliMeter) / 1000.0;
-  sensor_msg_array[i].header.stamp = ros::Time::now();
-  std::string frame = "sensor";
-  sensor_msg_array[i].header.frame_id = frame + std::to_string(i + 1);
-  sensor_msg_array[i].field_of_view = 0.436332;
-  sensor_msg_array[i].min_range = 0.03;
-  sensor_msg_array[i].max_range = 2.0;
+void thread(int i) {
```
Comment (bhuvanchandradv) [inline src/vl53l0x_driver.cpp]: readable names

File: src/vl53l0x_driver.cpp [hunk 5]
```diff
@@ -136,13 +143,13 @@ int main(int argc, char **argv) {
         sensor_pub_array[i] = nh.advertise<vl53l0x_driver::vl53l0x>(result, 10);
         Sensor_Calibration(pSensors[i]);
       }
-      while (ros::ok() and check_device_connection(pSensors[0]->fd)) {
-        for (int i = 0; i < NUM_SENSORS; i++) {
-          Start_Ranging(i);
-          sensor_pub_array[i].publish(sensor_msg_array[i]);
-          ros::spinOnce();
-        }
+      ros::AsyncSpinner spinner(NUM_SENSORS); // Use 4 threads
```
Comment (bhuvanchandradv) [inline src/vl53l0x_driver.cpp]: remove the comment

File: src/vl53l0x_driver.cpp [hunk 6]
```diff
@@ -136,13 +143,13 @@ int main(int argc, char **argv) {
...
-        for (int i = 0; i < NUM_SENSORS; i++) {
-          Start_Ranging(i);
-          sensor_pub_array[i].publish(sensor_msg_array[i]);
-          ros::spinOnce();
-        }
+      ros::AsyncSpinner spinner(NUM_SENSORS); // Use 4 threads
+      spinner.start();
+      for (int i = 0; i < NUM_SENSORS; i++) {
+        boost::thread(boost::bind(thread, i));
       }
```
Comment (bhuvanchandradv) [inline src/vl53l0x_driver.cpp:152]: Drop the extra line above

**Review by bhuvanchandradv — APPROVED** [review]

File: src/vl53l0x_driver.cpp (hunk 4 above)
Comment (tomin11) [reply src/vl53l0x_driver.cpp]: ok

**Pull request merged**